### ISSUE TITLE
3.6.0 smooth scrolling with hotkey is not smooth

### ISSUE THREAD

**Issue by poblabs** [Athou/commafeed#1087]
- CommaFeed version (or "commafeed.com"): 3.6.0
- Browser [e.g. chrome, firefox]: Chrome 114.0.5735.134
- Device [e.g. desktop, mobile]: Desktop, Windows 11

I am a long time user of self-hosted Commafeed 2.6.0 and I'm testing out 3.6.0 and the smooth scrolling is not smooth when using a hotkey. It's pretty choppy. 

I noticed it happens more when there are a lot of unread articles. Right now I have 120 unread articles, and pressing the J or N hotkey results in a choppy scroll experience. In 2.6.0 this was very smooth so I thought I'd report it.

**Comment by Athou** [comment]
It is indeed choppy with Chrome, thanks for reporting the issue.

**Comment by Athou** [comment]
Could you please try with this change please? It seems to help a lot here.
The jar is available at https://github.com/Athou/commafeed/actions/runs/5320555387

**Comment by poblabs** [comment]
It's better for sure! I had 1 bit of choppiness but that could be my Chrome. I'll have to give it some more testing throughout the day. 

Another question for you with the hotkey scrolling. Would it be possible to always scroll the next article to the top even if it fits within the view? 

For example in this screenshot, I hit `j` to go to the next unread but it didn't scroll my feed to the top so I keep thinking it didn't work. Again, slight difference between v2 and v3 that I noticed.  

![image](https://github.com/Athou/commafeed/assets/3484775/08d21bc2-d0c8-4ff6-ac77-e1405d3e860c)

**Comment by poblabs** [comment]
Sure thing - #1088 , thanks again!

**Comment by poblabs** [comment]
Compared to 2.0, I'm still having a bit choppy with CommaFeed version 3.7.0 (7c226f4). Could it be the speed it's scrolling? Is there an option to adjust scroll speed?

**Comment by Athou** [comment]
I profiled the app in Chrome and made some changes.
Does this help https://github.com/Athou/commafeed/actions/runs/5366267839 ?

**Comment by poblabs** [comment]
so far so good! the only thing now is there's a delay (maybe 750ms?) from the keypress to the scroll event.

**Comment by Athou** [comment]
I made more changes to help with the initial lag.
Could you test https://github.com/Athou/commafeed/actions/runs/5371503325 please?

**Comment by poblabs** [comment]
It's getting better! Compared to 2.0, still a bit slow but making progress to make it feel less laggy

**Comment by ehamiter** [comment]
> Is there an option to adjust scroll speed?

I am also curious about this. I would prefer to have no animation at all, and immediately cut to the next entry. In `commafeed-client/src/app/slices.ts`, I see this:

```typescript
if (arg.scrollToEntry) {
    const entryElement = document.getElementById(Constants.dom.entryId(entry))
    if (entryElement) {
        const alwaysScrollToEntry = state.user.settings?.alwaysScrollToEntry
        const entryEntirelyVisible = Constants.layout.isTopVisible(entryElement) && Constants.layout.isBottomVisible(entryElement)
        if (alwaysScrollToEntry || !entryEntirelyVisible) {
            const scrollSpeed = state.user.settings?.scrollSpeed
            thunkApi.dispatch(entriesSlice.actions.setScrollingToEntry(true))
            scrollToEntry(entryElement, scrollSpeed, () => thunkApi.dispatch(entriesSlice.actions.setScrollingToEntry(false)))
        }
    }
}
```
Can I set `alwaysScrollToEntry` or `scrollSpeed`? If so, where? 

Thanks for any info!

**Comment by Athou** [comment]
> Can I set `alwaysScrollToEntry` or `scrollSpeed`? If so, where?

Sure, those are the first two options in the user settings:

![image](https://github.com/Athou/commafeed/assets/1256795/d43c9890-3fac-4b4e-94f8-ded1fac2e834)

**Comment by ehamiter** [comment]
🤦 

@Athou thanks so much, I swear I looked through the settings a billion times but somehow never noticed those.

**Comment by DarkFox** [comment]
This seems to still be an issue in 5.9.0
In addition, if there's a lot of items in the feed, there's a significant delay when navigating between items.

**Comment by Athou** [comment]
> This seems to still be an issue in 5.9.0
> In addition, if there's a lot of items in the feed, there's a significant delay when navigating between items.

Indeed, I was able to reproduce the issue when a large number of entries are loaded, particularly in the expanded view.
I’ve pushed two new commits that should address this problem, especially the one enabling the React compiler. With these changes, the issue no longer occurs in either Firefox or Chrome.
The fix has been deployed to [commafeed.com](https://commafeed.com/) and will be included in the next release. Let me know how it works for you!
